Sources/JSON/JSON.Rule.swift

```diff
@@ -45,7 +45,7 @@ extension JSON
     ///     even when applied to third-party collection types, like 
     ///     ``/swift-nio/NIOCore/ByteBufferView``.
     public 
-    enum Rule<Location>
+    enum Rule<Location>:ParsingRule
```

Comment:
code with `JSON.Rule<Location>` implementing the root [`ParsingRule`](https://swiftinit.org/reference/swift-grammar/grammar/parsingrule): 

[**workflow run**](https://github.com/kelvin13/swift-json/runs/7618901839?check_suite_focus=true#step:7:52)
```
 Build complete! (37.48s)
swift-json: decoded 38295 messages
foundation: decoded 38295 messages
swift-json decoding time: 2.875409557 seconds
foundation decoding time: 3.795551295 seconds
```